src/swupd_internal.h

```diff
@@ -22,6 +25,59 @@ enum swupd_code repair_main(int argc, char **argv);
 enum swupd_code diagnose_main(int argc, char **argv);
 enum swupd_code bundle_info_main(int argc, char **argv);
 enum swupd_code verify_main(int argc, char **argv);
+enum swupd_code third_party_main(int argc, char **argv);
+
+/**
+ * @brief Creates a new third-party repo under THIRDPARTY_REPO_PREFIX
+ * if it does not exist
+ *
+ * @return 0 on success, SWUPD_THIRD_PARTY_REPO_EXISTS if it already exists
+ * and negative numbers on errors
+ */
+extern enum swupd_code third_party_add_main(int argc, char **argv);
```

Comment:
I had mentioned this in a previous comment but I could not find it anymore, but you cannot use `enum swupd_code` and return negative numbers on errors. The whole purpose of using `enum soup_code` is to guarantee the function is going to return SWUPD_OK or SWUPD_NO if successful, and any other error within the swupd_exit_codes on errors.

Reply:
found the comment I'd made, here it is:

"All these functions are defined as returning a value of enum swupd_code, which at the moment are positive values from 0 to 36.

When defining functions within the swupd code it is possible to define functions that return any kind of value, boolean, positive numbers, negative numbers, etc. However whenever swupd exits, it must always exit with a positive value contained in swupd_code. In other words all functions that return values different than swupd_code can only be used internally, and whatever function that is going to propagate its return code all the way to the user has to return a value defined within swupd_code.

The whole purpose of using the enum swupd_code as return type is to "certify" in a way that a certain function complies with only returning positive values within the range of swupd_code, and this way we know it is safe to propagate its return code all the way to the user. So by this definition it makes no sense to define a function that supposedly complies with only returning values within swupd_code range, but arbitrarily returning negative numbers on errors. If a function returns a swupd_code then it must return SWUPD_OK or SWUPD_NO in a successful operation, and an appropriate error from swupd_code on failures."

Reply:
Sorry about missing that. Will correct it in next push